source/material_model/visco_plastic.cc

```diff
@@ -252,12 +252,13 @@ namespace aspect
           double viscous_stress = 2. * viscosity_pre_yield * edot_ii;
 
           // Calculate Drucker Prager yield strength (i.e. yield stress)
+          // Uses max_yield_strength to limit the yield strength for slab beneath the lithosphere
           double yield_strength = ( (dim==3)
                                     ?
-                                    ( 6.0 * coh * std::cos(phi) + 6.0 * std::max(pressure,0.0) * std::sin(phi) )
-                                    / ( std::sqrt(3.0) * (3.0 + std::sin(phi) ) )
+                                    std::min(( 6.0 * coh * std::cos(phi) + 6.0 * std::max(pressure,0.0) * std::sin(phi) )
+                                    / ( std::sqrt(3.0) * (3.0 + std::sin(phi) ) ), max_yield_strength)
                                     :
-                                    coh * std::cos(phi) + std::max(pressure,0.0) * std::sin(phi) );
+                                    std::min(coh * std::cos(phi) + std::max(pressure,0.0) * std::sin(phi), max_yield_strength) );
```

Comment:
I would prefer if we make this not more compilcated than it already is. Maybe just put it in a new line:
```
double yield_strength = ....

yield_strength = std::min(yield_strength, max_yield_strength);
```

Reply:
Agree with @gassmoeller, the formulas are already quite long.